File: Codes/001_two_sum/1.两数之和.go
/*
 * @Author: Theo_hui
 * @Date: 2021-08-08 19:30:55
 * @Descripttion:
 */
/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

/*
	1.粗暴遍历法
*/
// @lc code=start
// func twoSum(nums []int, target int) []int {
// 	// 双下标
// 	index1, index2 := 0, 0
// 	for i := 0; i < len(nums)-1; i++ {
// 		for j := i + 1; j < len(nums); j++ {
// 			if nums[i]+nums[j] == target {
// 				index1, index2 = i, j
// 			}
// 		}
// 	}

// 	return []int{index1, index2}

// }

/*
	2.哈希法
*/
// @lc code=start
func twoSum(nums []int, target int) []int {
	//1. 创建哈希表
	hashT := map[int]int{}

	for i, v := range nums {
		if j, ok := hashT[target-v]; ok {
			return []int{i, j}
		}
		hashT[v] = i
	}

	return []int{}
}

// @lc code=end

File: Codes/001_two_sum/1.两数之和.py
'''
Author: Theo_hui
Date: 2021-08-08 20:06:43
Descripttion: 
'''
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
    # 1. 暴力遍历法
    # def twoSum(self, nums: List[int], target: int) -> List[int]:
    #     for i in range(len(nums)-1):
    #         for j in range(i+1,len(nums)):
    #             if nums[i]+nums[j]==target:
    #                 return [i,j]

    # 2. 哈希表法
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashT = {}

        for i,v in enumerate(nums):
            if target-v in hashT.keys():
                return [i,hashT[target-v]]
            else:
                hashT[v]=i
# @lc code=end

